awx/ui_next/Makefile

```diff
@@ -0,0 +1,97 @@
+## UI_NEXT_DIR: Relative path to the directory containing this Makefile
+UI_NEXT_DIR := $(patsubst %/,%,$(dir $(lastword $(MAKEFILE_LIST))))
+
+# ## Path to your local clone of the UI_NEXT repo
+# #  NOTE: you will not be able to build within the docker-compose development environment if you use this option
+UI_NEXT_LOCAL ?=
+
+# Git repo and branch to the UI_NEXT repo
+UI_NEXT_GIT_REPO ?= https://github.com/ansible/ansible-ui.git
+UI_NEXT_GIT_BRANCH ?= main
+
+.PHONY: ui-next
+## Default build target of ui-next Makefile, builds ui-next/build
+ui-next: ui-next/build
+
+.PHONY: ui-next/build
+## Build ui-next/build
+ui-next/build: $(UI_NEXT_DIR)/build
+
+## Real target for ui-next.
+$(UI_NEXT_DIR)/build: ui-next/src/build
+	@echo "=== Copying $(UI_NEXT_DIR)/src/build to $(UI_NEXT_DIR)/build ==="
+	@rm -rf $(UI_NEXT_DIR)/build
+	@cp -r $(UI_NEXT_DIR)/src/build $(UI_NEXT_DIR)
+	@echo "=== Done building $(UI_NEXT_DIR)/build ==="
+
+.PHONY: ui-next/src/build
+## Build ui-next/src/build
+ui-next/src/build: $(UI_NEXT_DIR)/src/build
+
+## Real target for ui-next/src/build.
+$(UI_NEXT_DIR)/src/build: ui-next/src ui-next/src/webpack
+	@echo "=== Building ui_next ==="
+	@cd $(UI_NEXT_DIR)/src && npm run build:awx
+
+.PHONY: ui-next/src
+ui-next/src: $(UI_NEXT_DIR)/src
+
+.PHONY: $(UI_NEXT_DIR)/src
+## Real target for ui-next/src.
+$(UI_NEXT_DIR)/src:
+	@echo "=== Setting up $(UI_NEXT_DIR)/src ==="
+	@if [ ! -z "$(UI_NEXT_LOCAL)" ]; then \
+		if [ -d $(UI_NEXT_DIR)/src ]; then \
+			if [ "$$(readlink $(UI_NEXT_DIR)/src)" = "$(UI_NEXT_LOCAL)" ]; then \
```

Comment:
Let's try not to use `readlink` so we're inclusive of BSD (and macOS, I guess).

Reply:
this one works for macOS... i verified with BSD readlink